pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -112,6 +115,26 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 		controllers.WithReconciler(c.onEvent),
 		controllers.WithMaxAttempts(5))
 	c.ingress.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+
+	// When the service which is referred by ingress using port name has been changed,
+	// we should trigger to rebuild rds.
+	c.services.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+		serviceName := config.NamespacedName(o).String()
+		// We only care about the service which is referred by ingress using port name.
+		if c.serviceStore.contains(serviceName) {
+			vsmetadata := config.Meta{
+				Name:             o.GetName() + "-" + "virtualservice",
```

Comment:
Previously we used ingress name as vs prefix

Reply:
Good catch. But this service may be referred by more than one ingresses. So here I choose to the service name. From my knowledge, the config name seems is useless currently. Or maybe you can give some suggestions. Thanks.